hydradx/tests/test_omnipool_amm.py

```diff
@@ -1679,4 +1679,298 @@ def test_oracle_one_block_with_swaps(liquidity: list[float], lrna: list[float],
+        if omnipool.last_fee[tkn] != pytest.approx(expected_asset_fee, rel=1e-15):
+            raise AssertionError('Asset fee is not correct.')
+
+
+@given(
+    st.lists(asset_quantity_strategy, min_size=3, max_size=3),
+    st.lists(asset_quantity_bounded_strategy, min_size=3, max_size=3),
+    st.lists(asset_quantity_strategy, min_size=3, max_size=3),
+    st.lists(asset_quantity_strategy, min_size=3, max_size=3),
+    st.lists(asset_quantity_strategy, min_size=3, max_size=3),
+    st.lists(asset_price_strategy, min_size=2, max_size=2),
+    st.integers(min_value=10, max_value=1000),
+    st.floats(min_value=-1000, max_value=1000),
+    st.lists(st.floats(min_value=0.0005, max_value=0.10), min_size=3, max_size=3),
+    st.lists(st.floats(min_value=0.0025, max_value=0.40), min_size=3, max_size=3),
+    st.lists(st.floats(min_value=0.001, max_value=100), min_size=2, max_size=2),
+    st.lists(st.floats(min_value=0.000001, max_value=0.0001), min_size=2, max_size=2),
+)
+def test_dynamic_fees_with_trade(liquidity: list[float], lrna: list[float], oracle_liquidity: list[float],
+                                 oracle_volume_in: list[float], oracle_volume_out: list[float],
+                                 oracle_prices: list[float], n, trade_size: float, lrna_fees: list[float],
+                                 asset_fees: list[float], amp: list[float], decay: list[float]):
+
+    init_liquidity = {
+        'HDX': {'liquidity': liquidity[0], 'LRNA': lrna[0]},
+        'USD': {'liquidity': liquidity[1], 'LRNA': lrna[1]},
+        'DOT': {'liquidity': liquidity[2], 'LRNA': lrna[2]},
+    }
+
+    init_oracle = {
+        'liquidity': {'HDX': oracle_liquidity[0], 'USD': oracle_liquidity[1], 'DOT': oracle_liquidity[2]},
+        'volume_in': {'HDX': oracle_volume_in[0], 'USD': oracle_volume_in[1], 'DOT': oracle_volume_in[2]},
+        'volume_out': {'HDX': oracle_volume_out[0], 'USD': oracle_volume_out[1], 'DOT': oracle_volume_out[2]},
+        'price': {'HDX': oracle_prices[0], 'USD': 1, 'DOT': oracle_prices[1]},
+    }
+
+    init_lrna_fees = {
+        'HDX': lrna_fees[0],
+        'USD': lrna_fees[1],
+        'DOT': lrna_fees[2],
+    }
+
+    init_asset_fees = {
+        'HDX': asset_fees[0],
+        'USD': asset_fees[1],
+        'DOT': asset_fees[2],
+    }
+
+    asset_fee_params = {
+        'minimum': 0.0025,
+        'amplification': amp[0],
+        'raise_oracle_name': 'oracle',
+        'decay': decay[0],
+        'fee_max': 0.4,
+    }
+
+    lrna_fee_params = {
+        'minimum': 0.0005,
+        'amplification': amp[1],
+        'raise_oracle_name': 'oracle',
+        'decay': decay[1],
+        'fee_max': 0.1,
+    }
+
+    initial_omnipool = oamm.OmnipoolState(
+        tokens=copy.deepcopy(init_liquidity),
+        oracles={
+            'oracle': n
+        },
+        asset_fee={
+            'HDX': dynamicadd_asset_fee(
```

Comment:
Since the asset fees are all the same, you don't have to specify them individually. Is there a reason why you do?

Reply:
Not particularly, I think I was planning on maybe randomizing them separately, but I am aware that that is not currently how it's implemented. I think my preference now would be to leave it as it is currently written, even though it's a little redundant. IMO a future task is to clean up how we instantiate global pool states for these tests. I know we have some logic that I'm not using, but if we are going to try to always test at the public API (i.e. by calling "run") then it is probably worth it to make it easy to instantiate the various objects needed to call that function.